addons/website/models/website.py

```diff
@@ -1033,7 +1033,7 @@ def _force_website(self, website_id):
             request.session['force_website_id'] = website_id and str(website_id).isdigit() and int(website_id)
 
     @api.model
-    def is_publisher(self):
+    def is_restricted_editor(self):
         return self.env['ir.model.access'].check('ir.ui.view', 'write', False)
```

Comment:
Was already the case before but strange that we have a method `is_XXX` which can return `True` even if you don't have the `group_XXX` no? :grimacing:

Reply:
Well, this method seems wrong actually.
It is not checking for restricted editor (or publisher before this PR and since its introduction with https://github.com/odoo/odoo/commit/1acf6d904fa8474b6ba59735ae57a0323465178d 9 years ago) but for full editor / designer.

Restricted editor / publisher don't have the write right on ir.ui.view, always been like that https://github.com/odoo/odoo/blob/1acf6d904fa8474b6ba59735ae57a0323465178d/addons/website/security/ir.model.access.csv

I guess there was already some confusion about the naming back then :shrug: 

I will have a look tomorrow, I guess we should rename it and make it check for the group instead of ir.ui.view write right :thinking:

Reply:
> Restricted editor / publisher don't have the write right on ir.ui.view, always been like that

Well, precisly but this is what the method is doing: check that you don't have the write access on ir.ui.view. So the method is probably used rightfully everywhere and we don't wan't to use has_group (although I guess we could, always been unclear when we want to do has_group or access check). But the name of the method is confusing: I guess that in all flows is_restricted_editor === has_group(restricted_editor) but technically I would expect that is_restricted_editor checks if you have the group as the name suggests.

Reply:
(Considering the code before this PR, so let's not think about the new naming `is_restricted_editor()`)
What the method does, despite its `is_publisher()` name is actually indirectly checking if you are a `website_designer` and not a `website_publisher` :grimacing: (on top of that it also allow you do be considered as designer if you have write access on view from another module, which doesn't makes sens actually, I don't see why someone from module xxx not related to website but which gives the ir.ui.view write would suddenly be considered as designer (wrongly "publisher" according to the wrong/misleading method name) and would see stuff supposed to be seen only by the website builder admin).

At the end, this method should just be renamed in to `is_designer()`, or even better, simply removed and replaced by call to `has_group(designer)`, as I don't see any valid reason for someone with write right on ir.ui.view be considered as a website builder admin :grimacing:

Reply:
Maybe we could also add a comment in stable version to mention that despite being named `is_publisher()`, it actually check if you are a `designer`, not a `publisher`.

Reply:
Ok I was reading the method wrong, hence the confusion. I somehow thought that `self.env['ir.model.access'].check('ir.ui.view', 'write', False)` was checking you *don't* have write access because of the misleading "False" x)

> At the end, this method should just be renamed in to is_designer(), or even better, simply removed 

Indeed, I would entirely remove it, it's useless

> and replaced by call to has_group(designer), as I don't see any valid reason [...]

I guess it has to be checked case by case, indeed someone with write access on views should not be considered a website builder admin but maybe that method was used in wrong cases not related to website builder, seems strange we would introduce that for no reason? Don't know :grimacing:

> Maybe we could also add a comment in stable version to mention that despite being named is_publisher(), it actually check if you are a designer, not a publisher.

Depending on what we do, I would add a deprecated comment indeed.